docs/blog/_posts/2018-10-10-10th-dotty-milestone-release.md

```diff
@@ -0,0 +1,248 @@
+
+### Java 9+
+
+Dotty now supports the latest versions of Java inlcuding Java 11!
+
+### Type-level programming: Match Types
+
+We've introduced a new form of types called match types. A match types is a mechanism for checking a
+type against a pattern. A match type reduces to one of a number of right hand sides, depending on a
+scrutinee type. E.g:
+
+```scala
+type Elem[X] = X match {
+  case String => Char
+  case Array[t] => t
+  case Iterable[t] => t
+}
+```
+
+This defines a type that, depending on the scrutinee type `X`, can reduce to one of its right hand
+sides. For instance,
+
+```scala
+Elem[String]       =:=  Char
+Elem[Array[Int]]   =:=  Int
+Elem[List[Float]]  =:=  Float
+Elem[Nil]          =:=  Nothing
+```
+
+Here `=:=` is understood to mean that left and right hand sides are mutually subtypes of each other.
+
+This feature is still experimental and subject to changes. For more information, visit the
+[Match Types](http://dotty.epfl.ch/docs/reference/match-types.html) section of our documentation.
+
+### Documentation in the REPL
+
+The previous release added documentation support for the IDE. Users can now query the documentation
+of sources previously compiled with Dotty within the REPL:
+
+```scala
+scala> /** An object */ object O { /** A def */ def foo = 0 }
+// defined object O
+
+scala> :doc O
+/** An object */
+
+scala> :doc O.foo
+/** A def */
+```
+
+### Tail-recursive methods can now be polymorphic
+
+Previously, a tail recursive call would be optimised only if the type arguments of the method
+or the enclosing class did not change at call site. E.g.
+
+```scala
+@tailrec def loop[T](x: T): Int = {
+  ...
+  loop[Int](1)
+}
+```
+
+```shell
+   loop[Int](1)
+   ^^^^^^^^^^^^
+  Cannot rewrite recursive call: it changes type arguments on a polymorphic recursive call
+```
+
+This restriction has now been removed. We also  improve upon `scalac` which is not able to optimise
+methods that change the type of `this` on a polymorphic recursive call.
+[Examples](https://github.com/lampepfl/dotty/blob/7a45a4a386d33180e5b7b21aa74271a77cce4707/tests/neg-tailcall/tailrec.scala#L43-L44)
+can be found in our test suite.
+
+### Experimental support for generic Tuples
+
+We augmented the `Tuple` class with generic methods such as `head`, `tail`, `apply`, `*:` and `++`.
+
+```scala
+scala> val t0 = (1, "2", 3L)
+val t0: (Int, String, Long) = (1,2,3)
+
+scala> val head = t0.head
+val head: Int = 1
+
+scala> val tail = t0.tail
+val tail: (String, Long) = (2,3)
+
+scala> val t1 = 0.0 *: t0
+val t1: (Double, Int, String, Long) = (0.0,1,2,3)
+
+scala> val t2 = t0 ++ t0
+val t2: (Int, String, Long, Int, String, Long) = (1,2,3,1,2,3)
+```
+
+### And much more!
+
+Please read our [release notes](https://github.com/lampepfl/dotty/releases/tag/0.10.0-RC1)
+for more details!
+
+## Breaking changes
+
+Dotty 0.10.0-RC1 requires sbt 1.2.3 and sbt-dotty 0.2.4.
+
+## Trying out Dotty
+
+### sbt
+
+Using sbt 1.1.5 or newer, do:
```

Comment:
"1.2.3 or newer" here, too

Reply:
Okay, but I think other readers will, like me, 1) be confused, 2) assume it was probably a mistake